torch/_dynamo/compiled_autograd.py

```diff
@@ -323,9 +422,13 @@ def move_graph_nodes_to_cuda(self, graph) -> List[int]:
             is_scalar = len(node.meta["val"].size()) == 0
             if is_cpu and is_scalar:
                 node_users = list(node.users.keys())
+                # We can only move the cpu scalar if it is not exposed to user code.
                 if all(
-                    isinstance(user.target, torch._ops.OpOverload)
-                    and user.target.namespace in ("prims", "aten")
+                    (
+                        isinstance(user.target, torch._ops.OpOverload)
+                        and user.target.namespace in ("prims", "aten")
+                    )
+                    or isinstance(user.target, torch._dynamo.compiled_autograd.Op)
```

Comment:
```suggestion
                    or isinstance(user.target, Op)
```
defined in the same file?